app/src/main/java/com/amaze/filemanager/filesystem/files/GenericCopyUtil.java

```diff
@@ -315,11 +315,16 @@ else if (outChannel != null) {
         if (bufferedOutputStream != null) copyFile(inChannel, bufferedOutputStream);
         else if (outChannel != null) copyFile(inChannel, outChannel);
       }
-
+      // till now file might not be written to disk yet, but to an OS buffer.
+      if (outputStream != null) outputStream.close();
```

Comment:
This might cause the stream not to be closed if the code before throws an exception.

Reply:
It will be closed anyways when GC collects the object. Exceptions are exceptional ;) So even though you may repeat it in finally block, i don't believe it really worth the complication. Especially since it doesn't cause much trouble even in the exceptional case.
The way it was, you were simply masked errors and reported that copy went fine, even when it wasn't. When you just do write, it doesn't mean that file was actually written. It means you have filled a buffer. Which only truly is flushed to disk at close (and not always even then). So only after checking the close, you can be confident that it actually reached the disk fine.

Reply:
This might silently leak memory for some implemetation of copy.

Reply:
@TranceLove opinion?

Reply:
If you need the outputStream to close unconditionally (if not null of course) you may want to do this in a `finally`.

Reply:
@dsvi Please move the code in such a way that it always executes.